packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -286,7 +286,10 @@ class FlutterDevice {
         ));
       }
     }
-    return vmService.onDone;
+    // TODO(jonahwilliams): this only seems to fail on CI.
+    return vmService.onDone.timeout(const Duration(seconds: 2), onTimeout: () {
+      globals.logger.printTrace('error: vm service shutdown failed');
```

Comment:
should just be for verifying that onDone did not return.

Reply:
Same here. If you feel you need to explain a line of code, consider leaving the explanation as a comment in the code.

Reply:
Should `onDone` have an `onError` handler as well that ignores the error?